### PR TITLE
shh changes

### PR DESCRIPTION
**Pull request by alexlitz**

### CODE REVIEW

File: shh/shh.go [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+//           to: DATA, 60 Bytes - (optional) Identity of the receiver. When present it 
+//              will try to decrypt any incoming message if the client holds the private key to this identity.
+//           topics: Array of DATA - Array of DATA topics which the incoming message's topics should match. 
+//              You can use the following combinations:
+//              [A, B] = A && B
+//              [A, [B, C]] = A && (B || C)
+//              [null, A, B] = ANYTHING && A && B null works as a wildcard
+// Returns:
+//       - QUANTITY - The newly created filter.
+func (ssh *SHH) NewFilter(to string, topics []string) {
```
Comment (regcostajr) [inline shh/shh.go]: It's supposed to be shh

Not have a the return types (string, error)

File: shh/shh.go [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+}
+
+// UninstallFilter - Uninstalls a filter with given id. Should always be called when watch is no longer needed. 
+//      Additonally Filters timeout when they aren't requested with shh_getFilterChanges for a period of time.
+// Reference: https://github.com/ethereum/wiki/wiki/JSON-RPC#shh_uninstallfilter
+// Parameters
+//       - QUANTITY - The filter id.
+// Returns:
+//       - Boolean - true if the filter was successfully uninstalled, otherwise false.
+func (ssh *SHH) UninstallFilter(id string) (bool, filter) {
```
Comment (regcostajr) [inline shh/shh.go]: It's supposed to be shh

"filter" on the return types, don't exists, i think do you mean (bool, error)

File: shh/shh.go [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+//              hash: DATA, 32 Bytes (?) - The hash of the message.
+//              from: DATA, 60 Bytes - The sender of the message, if a sender was specified.
+//              to: DATA, 60 Bytes - The receiver of the message, if a receiver was specified.
+//              expiry: QUANTITY - Integer of the time in seconds when this message should expire (?).
+//              ttl: QUANTITY - Integer of the time the message should float in the system in seconds (?).
+//              sent: QUANTITY - Integer of the unix timestamp when the message was sent.
+//              topics: Array of DATA - Array of DATA topics the message contained.
+//              payload: DATA - The payload of the message.
+//              workProved: QUANTITY - Integer of the work this message required before it was sent 
+func (ssh *SHH) GetFilterChanges(id string) []dto.WhisperMessage {
```
Comment (regcostajr) [inline shh/shh.go]: It's supposed to be shh

You are returning 2 , but just receiving 1, you have to add the error in the return types like: ([]dto.WhisperMessage, error)

File: shh/shh.go [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+//              hash: DATA, 32 Bytes (?) - The hash of the message.
+//              from: DATA, 60 Bytes - The sender of the message, if a sender was specified.
+//              to: DATA, 60 Bytes - The receiver of the message, if a receiver was specified.
+//              expiry: QUANTITY - Integer of the time in seconds when this message should expire (?).
+//              ttl: QUANTITY - Integer of the time the message should float in the system in seconds (?).
+//              sent: QUANTITY - Integer of the unix timestamp when the message was sent.
+//              topics: Array of DATA - Array of DATA topics the message contained.
+//              payload: DATA - The payload of the message.
+//              workProved: QUANTITY - Integer of the work this message required before it was sent 
+func (ssh *SHH) GetMessages(id string) []dto.WhisperMessage {
```
Comment (regcostajr) [inline shh/shh.go]: It's supposed to be shh

You are returning 2 , but just receiving 1, you have to add the error in the return types like: ([]dto.WhisperMessage, error)

**Comment by HarderThenBetter** [comment]
do you have method to listen event from contarct

**Pull request closed without merging**